assets/backend/store/appointments/appointments.ts

```diff
@@ -200,15 +231,28 @@ export const resolvers = {
 	): Generator<
 		FetchFromApiActionReturn,
 		{ type: string; appointments: Appointment[] },
-		APIResponse<{ appointments: Appointment[] }>
+		Response
 	> {
 		const response = yield baseActions.fetchFromAPI(
 			addQueryArgs('appointment', {
 				query,
 			})
 		);
 		const { data } = response;
-		const { appointments } = data;
-		return actions.setAppointments(appointments);
+		const {
+			appointments,
+			totalItems,
+			totalPages,
+			postsPerPage,
+			currentPage,
+		} = data;
+
+		return actions.setAppointments(
+			appointments,
+			totalItems,
+			totalPages,
+			postsPerPage,
+			currentPage
+		);
```

Comment:
Can we refactor this so `setAppointments` does not accept 5 arguments but object instead?

So here we could do this instead:

```ts
const { data } = response;
return actions.setAppointments(data);
```

And in the action instead of:

```ts
	setAppointments(
		appointments: Appointment[],
		totalItems: number,
		totalPages: number,
		postsPerPage: number,
		currentPage: number
	) {
		return {
			type: 'SET_APPOINTMENTS',
			appointments,
			totalItems,
			totalPages,
			postsPerPage,
			currentPage,
		} as const;
	},
```

We would do:

```ts
	setAppointments({
                appointments: Appointment[],
	        totalItems: number,
		totalPages: number,
		postsPerPage: number,
		currentPage: number   
	}) {
		return {
			type: 'SET_APPOINTMENTS',
			appointments,
		} as const;
	},
```